pkg/core/native/native_neo.go

```diff
@@ -600,9 +704,7 @@ func (n *NEO) ModifyAccountVotes(acc *state.NEOBalanceState, d dao.DAO, value *b
 		cd := new(candidate).FromBytes(si.Value)
 		cd.Votes.Add(&cd.Votes, value)
 		if !isNewVote {
-			if !cd.Registered && cd.Votes.Sign() == 0 {
-				return d.DeleteStorageItem(n.ContractID, key)
-			}
+			return n.checkCandidate(d, acc.VoteTo, cd)
```

Comment:
This breaks the logic, you're not supposed to `return` if the validator is still active.

Reply:
Fixed.